src/python/pants/pantsd/pantsd_integration_test.py

```diff
@@ -702,16 +705,19 @@ def test_unhandled_exceptions_only_log_exceptions_once(self):
 
         This is a regression test for the most glaring case of https://github.com/pantsbuild/pants/issues/7597.
         """
-        with self.pantsd_run_context(success=False) as ctx, temporary_dir(".") as directory:
-            Path(directory, "BUILD").write_text(
+        with self.pantsd_run_context(success=False) as ctx:
```

Comment:
This test doesn't fail, but the functionality of temporary_dir changed under the hood, so I kept it consistent in this file by using relative paths.

`tempfile.mkdtemp()` always return absolute path in 3.12+
https://docs.python.org/3/whatsnew/3.12.html#tempfile